File: python/soluc.py
import numpy as np
import time

def main():
    N = 1000

    A = np.random.rand(N, N)
    B = np.random.rand(N, N)

    start_time = time.time()

    result = np.dot(A, B)

    end_time = time.time()

    print(f"Tiempo de ejecución: {end_time - start_time:.4f}s")

if __name__ == "__main__":
    main()

File: javascript/soluc.js
console.time("Tiempo de ejecución");

// Tamaño de la matriz
const N = 1000;

// Generar dos matrices aleatorias de tamaño NxN
const A = Array.from({ length: N }, () => Array.from({ length: N }, () => Math.random()));
const B = Array.from({ length: N }, () => Array.from({ length: N }, () => Math.random()));

// Multiplicar las matrices
const result = Array.from({ length: N }, () => Array(N).fill(0));

for (let i = 0; i < N; i++) {
    for (let j = 0; j < N; j++) {
        for (let k = 0; k < N; k++) {
            result[i][j] += A[i][k] * B[k][j];
        }
    }
}

console.timeEnd("Tiempo de ejecución");